src/deploy.ts

```diff
@@ -97,13 +103,14 @@ async function execWithCredentials(
 }
 
 export async function deploy(gacFilename: string, deployConfig: DeployConfig) {
-  const { projectId, expires, channelId } = deployConfig;
+  const { projectId, expires, channelId, target } = deployConfig;
 
   const deploymentText = await execWithCredentials(
     "npx firebase-tools",
     [
       "hosting:channel:deploy",
       channelId,
+      !target ? "" : "--only " + target,
```

Comment:
```suggestion
      ...(target ? ["--only", target] : []),
```

(to match the `expires` line below)